win32/src/machine.rs

```diff
@@ -19,4 +21,24 @@ pub struct MachineX<Emu> {
     pub host: Box<dyn host::Host>,
     pub state: winapi::State,
     pub labels: HashMap<u32, String>,
+    pub exe_path: PathBuf,
+}
+
+pub enum StopReason {
+    /// The CPU is running normally.
+    None,
+    /// The CPU is blocked polling a future.
+    Blocked,
+    /// The CPU hit a debug breakpoint.
+    Breakpoint { eip: u32 },
+    /// The CPU hit a shim call.
+    ShimCall(&'static Shim),
```

Comment:
BTW, I have been (slowly) working on a change to how shims work that will require something very similar to this.

The summary is that kernel32.dll becomes a real dll that contains e.g.
```
_ReadFile:
  syscall
  ret 32
```

and then the cpu's syscall handler will trigger a StopReason like the above.

(The motivation for this change is (1) using a real dll makes it easier to handle exposing raw symbols from dlls as needed for msvcrt, and (2) a demoscene unpacker attempts to walk the dll headers in memory so I need actual dlls to exist in memory to appease them.)

Reply:
Coincidentally another win32 emu I've been following just made a similar change
https://inuh.net/@aaronsgiles@corteximplant.com/112889440660822739